src/lib.rs

```diff
@@ -1,10 +1,15 @@
 #![no_std]
 
+use core::convert::TryFrom;
 use core::mem::MaybeUninit;
 use core::ptr;
 
 extern crate libc;
 
+// avoid conflicting a real POSIX errno by using a value < 0
+// should we define this in ctru-sys somewhere or something?
```

Comment:
I wonder what the other libc function implementations found in libctru do (if they even set errno at all). If they don't, then it should be fine to just set here because these missing function implementations are the only ones who need to use this.

Reply:
Looking at the newlib patch [here](https://github.com/devkitPro/buildscripts/blob/master/dkarm-eabi/patches/newlib-4.2.0.20211231.patch), it seems like most of those implementations _do_ set `errno` as one would typically expect, whereas the [syscalls in libctru](https://github.com/devkitPro/libctru/blob/master/libctru/source/system/syscalls.c) mostly do nothing except the single call. Most of those are `void` functions, though, so maybe not the most representative set...

[This search](https://github.com/devkitPro/libctru/search?q=errno&type=code) shows most of the references to errno are in socket code, and use standard err numbers, except for `-socket_fd`. So... I'm not sure, all in all.

Also, this comment made me realize I forgot to remove the extra `ECTRU` declaration in there, so maybe in another patch set I can address it. I think your reasoning that we can leave it in this crate makes sense, though.